executor/load_data.go

```diff
@@ -113,17 +120,125 @@ type LoadDataInfo struct {
 	IgnoreLines uint64
 	Ctx         sessionctx.Context
 	rows        [][]types.Datum
+
+	//
+	commitTaskQueue chan commitTask
+	commitErrQueue  chan error
+	quitChan        chan struct{}
+}
+
+// GetRows getter for rows
+func (e *LoadDataInfo) GetRows() [][]types.Datum {
+	return e.rows
+}
+
+// GetCurBatchCnt getter for curBatchCnt
+func (e *LoadDataInfo) GetCurBatchCnt() uint64 {
+	return e.curBatchCnt
+}
+
+// CloseTaskQueue preparing routine to inform commit routine no more data
+func (e *LoadDataInfo) CloseTaskQueue() {
+	close(e.commitTaskQueue)
+}
+
+// ErrInCommitQueue check return value from commit routine
+func (e *LoadDataInfo) ErrInCommitQueue() bool {
+	return len(e.commitErrQueue) > 0
+}
+
+// GetErr get possible error from commit routine
+func (e *LoadDataInfo) GetErr() error {
+	var err error
+	if e.ErrInCommitQueue() {
+		err = <-e.commitErrQueue
+	}
+	return err
+}
+
+// InitQueues initialize task queue and error report queue
+func (e *LoadDataInfo) InitQueues() {
+	e.commitTaskQueue = make(chan commitTask, 1000)
+	e.commitErrQueue = make(chan error, 5)
+	e.quitChan = make(chan struct{}, 1)
+}
+
+// ForceQuitCommit let commit routine quit directly
+func (e *LoadDataInfo) ForceQuitCommit() {
+	e.quitChan <- struct{}{}
+}
+
+// EnqOneTask feed one batch commit task to commit routine
+func (e *LoadDataInfo) EnqOneTask() {
+	if e.curBatchCnt > 0 {
+		logutil.BgLogger().Info("tried to enqueue one task current len", zap.Uint64("len", e.curBatchCnt))
```

Comment:
We could make the committed batch size configurable. It is better to set it to debug level.